### ISSUE TITLE
[Feature Request] Please add Power controls to theme

### ISSUE THREAD

**Issue by wakatara** [3ximus/aerial-sddm-theme#34]
## TLDR

Besides the login and password for the greeter, it would be amazing if you could also add in stylin buttons for poweroff, reboot, hibernate, and sleep.

## Longer Tale

Fresh Arch install with sway and using this as the login theme before getting to minimalist sway on wayland. Matches with my macbook screensaver so kinda loving it. Thank you for doing this work!  Would love it though if it had power controls though to enhance the experience. 

Also, if you happen to know how to get the lockscreen to automatically kick in under sway wm, would love to hear it. Just dropped an issue on SDDM on this front. https://github.com/sddm/sddm/issues/1462 
gdm3's lockscreen kicks in under sway, but not sure why sddm's does not.

**Comment by 3ximus** [comment]
Hi,
Do you mean adding more buttons to the top of the screen (besides reboot and poweroff) ? Or the option to style them ?

SDDM is only a login manager and I doubt you will be able to find any solution for using it in a different way. You can try and copy my code into a QML theme and see if works (removing the login logic and so on)

Cheers

**Comment by wakatara** [comment]
Hmmm... Interestingly, I just shutdown and rebooted my machine.

At least in the initial login screen (one without video, just an image), I see the transparents session pulldown and a keyboard pulldown menu (note transparent and without labels, but do not see any buttons for poweroff and reboot at all (though I realize the vanilla SDDM has this). Is it possibly something with the theme? 

Note when I log out of sway and SDDM is brought back up as a video, no pulldowns nor buttons are presented to me at all. It is just the video playing and the username and password text fields.
(let me know if I have somehow failed to enable an option in the config I should have)

But yes, what I was asking for was that a button for both sleep and hibernate were added to the posssibilities of what options one has... much like you see in LightDM's display manager and in gdm3 (via pulldown menu on the far right.). 
thanks!

**Comment by wakatara** [comment]
As for locking, my understanding was that SDDM could take over the locking function so that was built into the DM as it is in LightDM (though I don't think anyone uses it) and gdm3 (which I think everyone does with gnome. I believe it also works with swaywm). 

Feel free to disabuse me of that notion if I somehow have that wrong though. I realize the DM session interaction with a window manager alone seems to be an complex subjet so can imagine the wm has to be dm aware in some way to known how to lock and deny access/input etc. For some strange reason I believed sddm and gdm did that. I might be mistaken.

**Comment by 3ximus** [comment]
> Hmmm... Interestingly, I just shutdown and rebooted my machine.
> 
> At least in the initial login screen (one without video, just an image), I see the transparents session pulldown and a keyboard pulldown menu (note transparent and without labels, but do not see any buttons for poweroff and reboot at all (though I realize the vanilla SDDM has this). Is it possibly something with the theme?
> 
> Note when I log out of sway and SDDM is brought back up as a video, no pulldowns nor buttons are presented to me at all. It is just the video playing and the username and password text fields. (let me know if I have somehow failed to enable an option in the config I should have)
> 
> But yes, what I was asking for was that a button for both sleep and hibernate were added to the posssibilities of what options one has... much like you see in LightDM's display manager and in gdm3 (via pulldown menu on the far right.). thanks!

The Poweroff + Reboot buttons should be there on the top right corner, they do appear for me...
However now the virtual keyboard always shows when I start typing, i think SDDM might have made some breaking changes on the latest updates.

Yeah I could add some extra buttons to the theme. I would like to make some visual changes to it to make the UI look more modern. I'll see if I have the patience to get back on this... 



> As for locking, my understanding was that SDDM could take over the locking function so that was built into the DM as it is in LightDM (though I don't think anyone uses it) and gdm3 (which I think everyone does with gnome. I believe it also works with swaywm).
> 
> Feel free to disabuse me of that notion if I somehow have that wrong though. I realize the DM session interaction with a window manager alone seems to be an complex subjet so can imagine the wm has to be dm aware in some way to known how to lock and deny access/input etc. For some strange reason I believed sddm and gdm did that. I might be mistaken.

Yeah, idk, I never looked that much into this subject, and these days I just use the default SDDM theme that comes with plasma that matches their Lock Screen...

Cheers
